### DISCUSSION TITLE
Array Filter with Handlebar Helpers

### DISCUSSION THREAD

**Discussion by Zero2XIII** [Budibase/budibase#8261] [Help] [answered]
Hi All,

I have an array with objects that have multiple properties, one of which is status.
I need to loop through them and bring back only those of certain status.
The final result I want to get to is to have a progress percentage on a line based on the linked elements.

```
Sample:

Task List [ 
{ Task: 1, Status: New},
{ Task: 2, Status: New},
{ Task: 3, Status: Complete},
]

End Result: 33% Complete
```

Unfortunately I have zero background in Javascript, and the handlebars documentation is confusing as it feels like it expects you to have some background knowledge when reading it.

I have tried so many variation but I can't seem to figure it out. Here is my pseudo code of what I am trying to achieve
```
{{ #filter Tasks Link Status!=Completed }}
   {{ Divide ( Length This ) ( Length Tasks Link ) }} % Completed
{{ else }}
  Error
{{ /filter }}
```

I will jump into the above with an if statement to prevent executing against a blank array.
This is for a "Formula" Column in my Job Cards table (Budibase Table)

I hope the above makes sense...

TYIA,
Philip

**Comment by andz-bb** [comment, marked as answer]
Hey @Zero2XIII 

I tried this with Handlebars and had some trouble with it myself. There may be a way to do it with the helpers we have but I wasn't able to see it. I have more experience with JS and found that using JS bindings is a much easier way to achieve what you're looking for.

In my example I have a data provider called Tasks that is linked to a table with the same name. I've set up the data in my table to be structured the same as in your example. Here's the JS I used to get the percentage.

```
const numberComplete = $("Tasks.Rows").filter(x => x.Status === "Complete").length;
const percentageComplete = (100 * numberComplete) / $("Tasks.Rows").length;
return `${Math.round(percentageComplete)}% complete
```

Hopefully this is fairly clear but if you would like any part of it explained in more detail let me know and I'll be happy to help 😊

**Reply by Zero2XIII** [reply]
Hi,

This seems to do the trick! Hopefully some edge cases won't be missed.
I really wish there were some documentation on JS and more detailed documentation on handle bars within budibase. The "No Code" is true up to this point and then I get lost super quick.

Best regards and thank you!

**Reply by NicolasFabre81** [reply]
ATM, for this kind of things, I prefer using custom queries to get the result, and then I only have to display it
